src/manifest/pythonEnvironment.ts

```diff
@@ -53,26 +66,63 @@ export class PythonEnvironment implements Disposable {
     this.executionDetails = await this.activatePythonExtension();
   }
 
-  private getPythonPathFromConfig(): string | undefined {
-    const value = workspace
-      .getConfiguration("dbt")
-      .get<string>("dbtPythonPathOverride");
-    return value ? substituteSettingsVariables(value) : undefined;
+  getResolvedConfigValue(key: string) {
+    const value = workspace.getConfiguration("dbt").get<string>(key, "");
+    return this.substituteSettingsVariables(value, this.environmentVariables);
   }
 
-  private parseEnvVarsFromUserSettings = (vsCodeEnv: {
-    [k: string]: string;
-  }) => {
+  private parseEnvVarsFromUserSettings = (
+    currEnvVars: EnvironmentVariables,
+    parentEnvVars: EnvironmentVariables,
```

Comment:
This seems very complicated for what we need to do. Why not simply only use `substituteSettingsVariables` on any new env var using the variable that are known at that time?